app/controllers/admin/salesforce_controller.rb

```diff
@@ -0,0 +1,27 @@
+module Admin
+  class SalesforceController < BaseController
+    layout 'admin'
+
+    def show
+    end
+
+    def callback
+      user = SalesforceUser.save_from_omniauth!(env["omniauth.auth"])
+      SalesforceUser.all.reject{|uu| uu.id == user.id}.each(&:destroy)
```

Comment:
If you think it should remain as an ActiveRecord table, though, you could do something like:

```
user_id = user.id
SalesforceUser.where{id != user_id}.delete_all
```